src/rrt/Tree.hpp

```diff
@@ -293,8 +308,15 @@ class Tree {
         Node<T>* best = nullptr;
 
         // k-NN search (O(log(N)))
-        flann::Matrix<double> query((double*)&state, 1,
-                                    sizeof(state) / sizeof(0.0));
+        flann::Matrix<double> query;
+        if (NULL == _TToArray) {
+            query = flann::Matrix<double>((double*)&state, 1,
+                                          sizeof(state) / sizeof(0.0));
+        } else {
+            double* data = new double[_dimensions];
```

Comment:
It's easiest to never use new if possible. It just makes everything hard. Could something like this work?
Query doesn't have to keep existing right?
```
vector<double> data(_dimensions);
flann::Matrix<double> query(data.data(), 1, sizeof(state) / sizeof(0.0));
```